catalog/core/catalog-core-subscriptionstore/subscriptionstore-impl/src/test/java/org/codice/ddf/catalog/subscriptionstore/SubscriptionContainerImplTest.java

```diff
@@ -0,0 +1,516 @@
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.subscriptionstore;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyObject;
+import static org.mockito.Matchers.anySet;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.RETURNS_DEEP_STUBS;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyZeroInteractions;
+import static org.mockito.Mockito.when;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Spliterator;
+
+import javax.cache.Cache;
+import javax.cache.CacheException;
+
+import org.codice.ddf.catalog.subscriptionstore.common.CachedSubscription;
+import org.codice.ddf.catalog.subscriptionstore.common.SubscriptionMetadata;
+import org.codice.ddf.catalog.subscriptionstore.internal.MarshalledSubscription;
+import org.codice.ddf.catalog.subscriptionstore.internal.SubscriptionFactory;
+import org.codice.ddf.catalog.subscriptionstore.internal.SubscriptionIdentifier;
+import org.codice.ddf.catalog.subscriptionstore.internal.SubscriptionStoreException;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Answers;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.osgi.framework.BundleContext;
+
+import com.google.common.collect.ImmutableList;
+
+import ddf.catalog.event.Subscription;
+
+@RunWith(MockitoJUnitRunner.class)
+public class SubscriptionContainerImplTest {
+
+    private static final String SOME_ID = "id";
+
+    private static final String SOME_TYPE = "type";
+
+    private static final String SOME_FILTER = "filter";
+
+    private static final String SOME_CALLBACK = "http://localhost8993/test";
+
+    @Mock(answer = Answers.RETURNS_DEEP_STUBS)
+    private static BundleContext mockBundleContext;
+
+    @Mock
+    private static Cache<String, CachedSubscription> mockCache;
+
+    @Mock
+    private static SubscriptionFactory mockFactoryA;
+
+    @Mock
+    private static SubscriptionFactory mockFactoryB;
+
+    @Mock
+    private Subscription mockArgSubscription;
+
+    @Mock
+    private SubscriptionIdentifier mockArgIdentifier;
+
+    @Mock
+    private MarshalledSubscription mockArgMarshalled;
+
+    private static Map<String, SubscriptionFactory> factories = new HashMap<>();
+
+    private SubscriptionContainerImpl container;
+
+    @Before
+    public void setup() {
+        when(mockArgIdentifier.getId()).thenReturn(SOME_ID);
```

Comment:
If your SubscriptionIdentifier was a class instead of an interface, you wouldn't need to mock it.